src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java

```diff
@@ -1504,4 +1508,393 @@ public void testValidateAnomalyDetectorWithWrongCategoryField() throws Exception
+                searchTopAnomalyResults(detector.getDetectorId(), false, "{\"start_time_ms\":1, \"end_time_ms\":2, \"size\":-1}", client());
+            }
+        );
+        assertTrue(negativeSizeException.getMessage().contains("Size must be a positive integer"));
+
+        // Zero size field
+        Exception zeroSizeException = expectThrows(
+            IOException.class,
+            () -> {
+                searchTopAnomalyResults(detector.getDetectorId(), false, "{\"start_time_ms\":1, \"end_time_ms\":2, \"size\":0}", client());
+            }
+        );
+        assertTrue(zeroSizeException.getMessage().contains("Size must be a positive integer"));
+
+        // Too large size field
+        Exception tooLargeSizeException = expectThrows(
+            IOException.class,
+            () -> {
+                searchTopAnomalyResults(
+                    detector.getDetectorId(),
+                    false,
+                    "{\"start_time_ms\":1, \"end_time_ms\":2, \"size\":9999999}",
+                    client()
+                );
+            }
+        );
+        assertTrue(tooLargeSizeException.getMessage().contains("Size cannot exceed"));
+
+        // No existing task ID for detector
+        Exception noTaskIdException = expectThrows(
+            IOException.class,
+            () -> { searchTopAnomalyResults(detector.getDetectorId(), true, "{\"start_time_ms\":1, \"end_time_ms\":2}", client()); }
+        );
+        assertTrue(noTaskIdException.getMessage().contains("No historical tasks found for detector ID " + detector.getDetectorId()));
+
+        // Invalid category fields
+        Exception invalidCategoryFieldsException = expectThrows(IOException.class, () -> {
+            searchTopAnomalyResults(
+                detector.getDetectorId(),
+                false,
+                "{\"start_time_ms\":1, \"end_time_ms\":2, \"category_field\":[\"invalid-field\"]}",
+                client()
+            );
+        });
+        assertTrue(
+            invalidCategoryFieldsException
+                .getMessage()
+                .contains("Category field invalid-field doesn't exist for detector ID " + detector.getDetectorId())
+        );
+
+        // Using detector with no category fields
+        AnomalyDetector detectorWithNoCategoryFields = createAnomalyDetector(
+            TestHelpers
+                .randomAnomalyDetectorUsingCategoryFields(
+                    randomAlphaOfLength(10),
+                    "timestamp",
+                    ImmutableList.of(indexName),
+                    ImmutableList.of()
+                ),
+            true,
+            client()
+        );
+        Exception noCategoryFieldsException = expectThrows(
+            IOException.class,
+            () -> {
+                searchTopAnomalyResults(
+                    detectorWithNoCategoryFields.getDetectorId(),
+                    false,
+                    "{\"start_time_ms\":1, \"end_time_ms\":2}",
```

Comment:
nit: maybe we should pass category field here just to make it clear the issue is that the detector itself has no category field and not that search request is missing it.

Reply:
The category field doesn't have to be passed in the request - the API will default to using all category fields found for the detector. I didn't want to pass a dummy category field value in the request since it would guarantee failure due to invalid category field, rather than missing category field from the detector itself. I tried to clarify the test case on line 1630.

Reply:
This makes sense to me! Thank you!